src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -107,31 +109,49 @@ public String deploy(AppDeploymentRequest request) {
 
 
 	@Override
-	public void undeploy(String id) {
-
-		String appId = fixKubeId(id);
+	public void undeploy(String appId) {
 		logger.debug("Undeploying module: {}", appId);
 
-		Map<String, String> selector = new HashMap<>();
-		selector.put(SPRING_APP_KEY, appId);
 		try {
-			client.services().withLabels(selector).delete();
-			client.replicationControllers().withLabels(selector).delete();
-			client.pods().withLabels(selector).delete();
+			if ("LoadBalancer".equals(client.services().withName(appId).get().getSpec().getType())) {
+				Service svc = client.services().withName(appId).get();
+				int tries = 0;
+				while (tries++ < 30) {
+					if (svc.getStatus().getLoadBalancer() != null &&
+							svc.getStatus().getLoadBalancer().getIngress() != null &&
+							svc.getStatus().getLoadBalancer().getIngress().isEmpty()) {
+						logger.debug("Waiting for LoadBalancer, try {}", tries);
```

Comment:
i'm wondering if maybe all of these debugs should be infos.....  or at least some of them to more easily see what is going on at the standard info level.